mono/mini/mini-runtime.c

```diff
@@ -2804,6 +2806,7 @@ mono_jit_runtime_invoke (MonoMethod *method, void *obj, void **params, MonoObjec
 		if (!is_ok (error))
 			return NULL;
 	} else {
+		MONO_SCOPE_ENABLE_JIT_EXEC();
```

Comment:
It seems odd this is needed in `mono_jit_runtime_invoke`

Reply:
Without this, JITted code won't be marked executable and you'll get EXC_BAD_ACCESS when trying to jump to it.

Reply:
Why don't we mark it as executable after JIT'ing instead of before executing?

Reply:
Executable mode is thread local state. That means changing the mode affects it for all code that runs on current thread. If we were to switch to executable mode right after JIT, we might mess up the state of something else running on the same thread (for instance, what if somebody embeds CEF?). Therefore we scope it: just before needing certain mode, we switch to it. And then after, we switch back to whatever it was.